zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/ResourceController.java

```diff
@@ -58,6 +60,18 @@ public CompletableFuture<ResponseEntity<Object>> deleteResource(
         .fold(RestErrorMapper::mapProblemToCompletedResponse, this::delete);
   }
 
+  @CamundaGetMapping(path = "/resources/{resourceKey}")
+  public CompletableFuture<ResponseEntity<Object>> getResource(
+      @PathVariable final long resourceKey) {
+    return getResource(new ResourceFetchRequest(resourceKey));
+  }
+
+  @CamundaGetMapping(path = "/resources/{resourceKey}/content")
+  public CompletableFuture<ResponseEntity<Object>> getResourceContent(
+      @PathVariable final long resourceKey) {
+    return getResourceContent(new ResourceFetchRequest(resourceKey));
+  }
```

Comment:
🔧 The methods called by these methods look a lot alike. There's some re-use possible:

```suggestion
  @CamundaGetMapping(path = "/resources/{resourceKey}")
  public CompletableFuture<ResponseEntity<Object>> getResource(
      @PathVariable final long resourceKey) {
    return RequestMapper.executeServiceMethod(
        () -> fetchResource(resourceKey), ResponseMapper::toGetResourceResponse);
  }

  @CamundaGetMapping(path = "/resources/{resourceKey}/content")
  public CompletableFuture<ResponseEntity<Object>> getResourceContent(
      @PathVariable final long resourceKey) {
    return RequestMapper.executeServiceMethod(
        () -> fetchResource(resourceKey), ResponseMapper::toGetResourceContentResponse);
  }

  private CompletableFuture<ResourceRecord> fetchResource(final long resourceKey) {
    return resourceServices
        .withAuthentication(RequestMapper.getAuthentication())
        .fetchResource(new ResourceFetchRequest(resourceKey));
  }
```

Reply:
Yep, good idea 👍🏻